src/editor/tools/place_critter_tool.cc

```diff
@@ -118,18 +118,13 @@ std::string EditorPlaceCritterTool::format_conf_description_impl(const ToolConf&
 }
 
 bool EditorPlaceCritterTool::save_configuration_impl(ToolConf& conf) {
-	int j = get_nr_enabled();
-
-	if (j == 0) {
+	if (0 == get_nr_enabled()) {
 		return false;
 	}
 
-	for (int i = 0; j > 0; ++i) {
-		if (is_enabled(i)) {
-			conf.map_obj_types.push_back(i);
-			--j;
-		}
-	}
+        for (int32_t descrindex: getEnabled()) {
+                conf.map_obj_types.insert(descrindex);
+        }
```

Comment:
I believe an insert-all can be written like this:
```suggestion
        conf.map_obj_types.insert(get_enabled().begin(), get_enabled().end());
```
Same in the other files that use this